hack/update-site-config.sh

```diff
@@ -3,11 +3,207 @@
+        if .systemd?.units then
+            # Delete prestaging service-units, including legacy content
+            del( .systemd.units[]
+                | select(
+                    .name == "precache-images.service"
+                    or .name == "precache-ocp-images.service"
+                    or ( .name == "var-mnt.mount" and (.contents | test ("BindsTo=precache")) )
+                    )
+                )
+        else . end
 
-BOOT_IGN="${REPODIR}/docs/resources/boot-beauty.ign"
-DISCOVERY_IGN="${REPODIR}/docs/resources/discovery-beauty.ign"
+        | if .storage?.files then
+            # Delete prestaging files, including legacy content
+            del( .storage.files[]
+                | select(
+                    .path == "/usr/local/bin/extract-ai.sh"
+                    or .path == "/usr/local/bin/extract-ocp.sh"
+                    or .path == "/usr/local/bin/agent-fix-bz1964591"
+                    )
+                )
+        else . end
+
+        # Delete empty objects that remain
+        | walk(
+            if type == "object" then
+                with_entries(if .value == {} or .value == [] then empty else . end)
+            else .  end)
+
+        # Delete entries where the only thing left is the "ignition" data
+        | del(select(keys == ["ignition"]))
+
+        | if type == "null" then empty else . end
+
+        end
+    '
+}
+
+#
+# Merge ignition config
+#
+function merge_ignition_config {
+    local field="${1}"
+    local ign="${2}"
+
+    (
+        yq "${field}" "${SITECFG}" | delete_prestaging_entries
+        sed -e "s/--label data/--label ${PARTITION_LABEL}/g" "${ign}"
+    ) \
+    | jq -csM '
+        if length == 1 then
+            # Only one element means there were no entries in the original
+            # site-config (or all have been pruned), so just return the prestaging config
+            .[0]
+        else
+            # Append the prestaging systemd units to the original site-config
+            if .[0].systemd then
+                if .[0].systemd.units then
+                    .[0].systemd.units += .[1].systemd.units
+                else
+                    .[0].systemd.units = .[1].systemd.units
+                end
+            else
+                .[0].systemd = .[1].systemd
+            end
+            # Append the prestaging storage files to the original site-config
+            | if .[0].storage then
+                if .[0].storage.files then
+                    .[0].storage.files += .[1].storage.files
+                else
+                    .[0].storage.files = .[1].storage.files
+                end
+            else
+                .[0].storage = .[1].storage
+            end
+            | .[0]
+        end
+    '
+}
+
+#
+# Update existing installer args
+#
+function update_installer_args {
+    local installer_args='"--save-partlabel","'"${PARTITION_LABEL}"'"'
+    if ! yq "${NODE_INSTALLER_ARGS}" "${SITECFG}" | jq -cM | grep "${installer_args}"; then
```

Comment:
Would it be  safer to use jq to check if the option exists? Just in case the installer_args contain special characters

Reply:
Do you have some example in mind? As far as I know, installerArgs should be a json array of args passed to the installer. Grepping the compact format of this json array for the desired pair of args should be safe, I would think (although this could/should be "grep -q" here).

Reply:
I can't think of any examples with a special character that couldn't be handled by grep. This is a more general comment since the script deals with JSON data rather than text files. However, considering the specific use case and the nature of the data being processed, I'm fine with using grep in this context.